### ISSUE TITLE
1.0.10 no email to agent & duplicate tickets

### ISSUE THREAD

**Issue by laurentiu2** [uvdesk/core-framework#299]
**Description**  
after creating a ticket (via email or web), an email should be sent to customer and another one to agent. Just the email to the customer is sent.
After creating the ticket via web, an error is displayed, although the ticket is created correctly.
I've tried all possibilities: response performing agent, assigned agent, uvdesk owner, none is getting the email.

**Additional context**  
![Screenshot 2020-05-10 at 14 51 29](https://user-images.githubusercontent.com/29857699/81498491-000d2080-92ce-11ea-801b-a273678d03d4.png)
![Screenshot 2020-05-10 at 11 39 10](https://user-images.githubusercontent.com/29857699/81498492-01d6e400-92ce-11ea-9781-3d0788de0821.png)

**Comment by papnoisanjeev** [comment]
@laurentiu2 

You can check the error by changing your environment to  **dev**  [here].(https://github.com/uvdesk/community-skeleton/blob/master/.env#L17)

**Comment by laurentiu2** [comment]
Changed to dev. Tried to create ticket via web.
It shows:
![Screenshot 2020-05-10 at 16 45 56](https://user-images.githubusercontent.com/29857699/81500952-1cb15480-92de-11ea-9455-4880062995c2.png)
![Screenshot 2020-05-10 at 16 47 28](https://user-images.githubusercontent.com/29857699/81500955-220e9f00-92de-11ea-9d53-7aea23aad7c5.png)

**Comment by vaishaliwebkul** [comment]
Hello @laurentiu2 
Thanks for notifying this bug.
Could you please mention your system environment?

- PHP Version

- Mysql Version 

- Composer Version

**Comment by papnoisanjeev** [comment]
@laurentiu2 
As we have checked after setting the workflow we are getting email on customer as well as agent email id.
Please specify your case what else you have done.

**Comment by laurentiu2** [comment]
> Could you please mention your system environment?
> 
>     * PHP Version
7.2.3
>     * Mysql Version
5.7 on a different server
>     * Composer Version
1.6.3

I installed UVdesk 1.0.10 on the same server where I have another instance of uvdesk (1.0.5 which is working fine)
I downloaded the zip from github, made a new database and all settings for apache. Nothing special.
On the previous install I had a similar problem, but the email was not sent to customer (see #294)

**Comment by papnoisanjeev** [comment]
@laurentiu2 

We need to check your panel as we are not able to reproduce this bug in our case everything working as expected.

Create a support ticket [here](https://support.uvdesk.com/en/customer/create-ticket/).

**Comment by laurentiu2** [comment]
It is getting stranger, I'll show step by step:
1. Customer sends an email to create a new ticket. Mail is received, ticket no.23 gets created. The customer receives the confirmation email, the agent doesn't.
![Screenshot 2020-05-11 at 15 45 33](https://user-images.githubusercontent.com/29857699/81563116-85600600-939e-11ea-9818-d00ef8163dbb.png)
![Screenshot 2020-05-11 at 15 46 25](https://user-images.githubusercontent.com/29857699/81563185-9ad53000-939e-11ea-8c84-52881a424d2e.png)

2. Agent respond to ticket. The customer receives the email.
![Screenshot 2020-05-11 at 15 49 50](https://user-images.githubusercontent.com/29857699/81563567-1636e180-939f-11ea-9c3b-51d1a1bba994.png)

3. Customer responds back.
The new response from customer gets into ticket 23.
The agent gets the email 'New response was added..' AND THEN
ticket 24 gets created with the same content as ticket 23
agent gets the email that ticket 24 was created.
![Screenshot 2020-05-11 at 15 54 23](https://user-images.githubusercontent.com/29857699/81564265-28fde600-93a0-11ea-97e3-c60eb71689d9.png)
![Screenshot 2020-05-11 at 15 55 07](https://user-images.githubusercontent.com/29857699/81564283-2f8c5d80-93a0-11ea-9d1d-c719b8ab39e2.png)


Something is gone totally wrong..

**Comment by laurentiu2** [comment]
> Create a support ticket here.
Maybe someone in the future finds this conversation useful here on github.

**Comment by papnoisanjeev** [comment]
@laurentiu2 

Is this issue you are facing in the web interface also? or just in email conversation.

**Comment by laurentiu2** [comment]
The first issue -agent is not receiving the email when new ticket is created- is also present in web interface.
The email conversation is particularly important in this scenario, we expect a high volume of emails.

**Comment by laurentiu2** [comment]
Is there anything I can check to see if everything had installed successfully?

**Comment by laurentiu2** [comment]
I have reinstalled using the zip from official website, not github.
The database would not work after setup, I had to manually configure it using 'php bin/console uvdesk:configure-helpdesk'
I also had to modify the correct url in config/packages/uvdesk.yaml

The tickets are ok, responses ok, all the emails sent correctly this time.
Either is a difference between .zip from github and .zip from official website or it is something else..

So, all good so far. Keep up the good work and congrats for a beautiful product!

**Comment by laurentiu2** [comment]
Nope, it is not working.
For no reason, the ticket gets created again after a few minutes (when the cron job is verifying the email)
See below:
![Screenshot 2020-05-12 at 08 37 54](https://user-images.githubusercontent.com/29857699/81642440-1204d580-942c-11ea-917a-8cd94d907804.png)

**Comment by laurentiu2** [comment]
After two or three re-installs, here are the conclusions:

1. the emails are sent correctly ONLY if email templates are not modified; which is impossible because one has to  customize all these responses. After modifying the templates the agent does not get anymore the email when a new ticket is created.
2. The duplication of tickets appears only after THE FIRST REPLY OF CUSTOMER TO AGENT RESPONSE (which is: customer initial message > agent reply to customer > customer reply to agent - after this step the duplication occurs)

It's impossible to use it in this state.

**Comment by papnoisanjeev** [comment]
@laurentiu2 
we will check and test also possible error if exist in system and try to resolve soon.

**Comment by laurentiu2** [comment]
I found out that the first issue (no email to agent) only appears when there's nothing in database! So only for the first ticket that gets created the agent does not get an email - and only after modifying the email templates. If I leave at least one ticket in the database, the agent does receive the email.

**Comment by papnoisanjeev** [comment]
> The duplication of tickets appears only after THE FIRST REPLY OF CUSTOMER TO AGENT RESPONSE (which is: customer initial message > agent reply to customer > customer reply to agent - after this step the duplication occurs)

We have followed your step but we did't face this issue please check once if you have done something different. Check Screenshot.

Ticket list:

![TicketsListMailtrail](https://user-images.githubusercontent.com/42460055/81899448-7bc1e280-95d8-11ea-8eb3-5e30796e6d13.png)

Ticket View:

![_68_Test_Mail_trail](https://user-images.githubusercontent.com/42460055/81899494-985e1a80-95d8-11ea-90d5-fe08d17cca27.png)


> the emails are sent correctly ONLY if email templates are not modified; which is impossible because one has to customize all these responses. After modifying the templates the agent does not get anymore the email when a new ticket is created.

Please let us know which email templates you have modified and what you have modified in email template, after which you are not able to receive notification for agent.

Please let us know required information so that we can check if there is any  issue exist with this.

**Comment by laurentiu2** [comment]
this is incredible..
> I found out that the first issue (no email to agent) only appears when there's nothing in database! So only for the first ticket that gets created the agent does not get an email - and only after modifying the email templates. If I leave at least one ticket in the database, the agent does receive the email.

So this issue is kinda solved. The other problem is duplication of tickets.

I have modified 'Customer Account Created' 'Ticket generated success mail to customer' and 'Ticket generated by customer' I have removed from the body {%global.companyName%} and {%global.companyLogo%} and modified the text.

For me the result is:
![Screenshot 2020-05-14 at 10 37 47](https://user-images.githubusercontent.com/29857699/81906978-c2f7a580-95cf-11ea-9ae9-92fdee2c8e67.png)
If I remove the duplicates, they keep creating once more. 
And all the emails are sent like they are new ones! That is the customer receives emails, the agent.

It has to be something related to how the app is working with the database.
For some reason it seems that it cannot verify that all those emails are already created as tickets.

**Comment by papnoisanjeev** [comment]
@laurentiu2 

Locate this path in your project  **vendor->uvdesk->mailbox-component->Services->MailboxService.php**

Share your **MailboxService.php** with us.

**Comment by laurentiu2** [comment]
[MailboxService.php.zip](https://github.com/uvdesk/core-framework/files/4629373/MailboxService.php.zip)

**Comment by papnoisanjeev** [comment]
@laurentiu2 
Replace your MailboxService.php code from [here](https://github.com/uvdesk/mailbox-component/edit/master/Services/MailboxService.php).

And we have updated the code here for remove ticket duplicate's as reported by you.
let us know if it solve your problem.

Replace your MailboxService.php code from given URl for you.

This change will reflect in next release or other can do the same by replacing  this file in their project till the next release.

**Comment by papnoisanjeev** [comment]
@laurentiu2 

Is this issue resolved or you need any assistance?

**Comment by laurentiu2** [comment]
Sorry for the delay. 
I just installed it and seems ok. If something goes wrong I'll ask for help.
Thanks.

**Comment by papnoisanjeev** [comment]
@laurentiu2 

We  have also released a new version of **community-skeleton v1.0.11** yesterday.
You can check it as many other issue also fixed in this version.

**Comment by laurentiu2** [comment]
@papnoisanjeev 
is it possible to upgrade somehow and not to make a new install? 
I installed so many times v1.0.10 I really don't have enough courage to do it again :)) for a few weeks

**Comment by papnoisanjeev** [comment]
@laurentiu2 

Yes, you can update all dependent  bundles in existing project using below commands then you don't need to create new project.

**composer update uvdesk/core-framework
composer update uvdesk/support-center-bundle
composer update uvdesk/mailbox-component
composer update uvdesk/automation-bundle
composer update uvdesk/extension-framework**

**Note:** Most probably your **swiftmailer.yaml, uvdesk_mailbox.yaml, uvdesk.yaml** **(under config->packages)** file content will become as it is in default installation so  if you **don't want to set your site_url , swiftmailer and mailbox** again then take a  backup of these file before running these command otherwise you need to just again set your **site_url**, **swiftmailer and mailbox.**
